osh/builtin.py

```diff
@@ -1031,6 +1031,12 @@ def _ParsePath(path_val):
 
 def _ResolveFile(name, path_list):
   # Now look for files.
+  # Relative path, won't be found by looking in PATH
+  if name.startswith('./') or name.startswith('../'):
```

Comment:
What happens when we run `subdir/myprogram` without `./` ?

I think there are probably 2 bugs.  One is a bug in `type` and the other is a bug in path lookup for execution.  We might need two different functions.

- $PATH shouldn't behave like the current directory is in it
- only programs directly under $PATH are run

Reply:
Hmm.. I want to keep this in the same function so that we don't have to duplicate the logic and we can make sure they're the same (`command -v` shouldn't print something different than what will be executed!). What if I changed the check to `name.contains('/')`? That won't look in the current directory unless you prefix it with `./` but it will still check subdirectories